spec/spec_helper.rb

```diff
@@ -39,7 +39,7 @@
   end
 
   config.before do
-    ActiveRecord::Base.clear_all_connections!
+    ActiveRecord::Base.connection_handler.clear_all_connections!
```

Comment:
Was this seriously the only change needed to get it working on 7.1+?

Have you tested it IN a rails application to ensure if fully works in 7.1+? (not wanting to just trust the tests here).

Reply:
I plan to test today in a Rails application. It definitely seemed weird but I noticed the code (line 14 in candidate.rb) was already using the connection handler so it seemed plausible it will work. The older AR versions would be supported with maybe some careful changes to their gemfiles. I'll keep you posted.